src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -387,6 +389,7 @@ public void setReadOnly(boolean readOnly) throws SQLException
    {
       delegate.setReadOnly(readOnly);
       isReadOnly = readOnly;
+      isCommitStateDirty = false;
```

Comment:
This change is valid.  The JavaDoc states that this method cannot be called during a transaction, therefore if the ``delegate.setReadOnly()`` call does not throw an exception, we can implicitly assume that the transaction was committed.  Therefore, it is same to imply that ``isCommitStateDirty`` is false.